File: python/phone-number/phone_number.py
import re


class PhoneNumber:
    def __init__(self, number):
        match = re.match(
            r'\D*1?\D*([2-9]\d{2})\D*([2-9]\d{2})\D*(\d{4})\D*$', number)

        if not match:
            raise ValueError("Invalid phone number")

        self.area_code, self.exchange_code, self.subscriber_number = match.groups()
        self.number = self.area_code + self.exchange_code + self.subscriber_number

    def pretty(self):
        return f'({self.area_code}) {self.exchange_code}-{self.subscriber_number}'

File: typescript/phone-number/phone-number.ts
export default class PhoneNumber {
  private static readonly RE: RegExp = /^(?:\+?1)?\s*\(?(\d{3})\)?[\s\.-]*(\d{3})[\s\.-]*(\d{4})\s*$/;

  constructor(private phoneNumber: string) {}

  number(): string | undefined {
    const match = this.phoneNumber.match(PhoneNumber.RE);
    return match ? `${match[1]}${match[2]}${match[3]}` : undefined;
  }
}
